src/tdb_decode.c

```diff
@@ -22,8 +23,22 @@ static int event_satisfies_filter(const tdb_item *event,
             return 0;
 
         while (i < next_clause){
-            uint64_t is_negative = filter[i++];
+            uint64_t comp_op = filter[i++];
             uint64_t filter_item = filter[i++];
+
+            /* Time range queries */
+            if (comp_op & TIME_RANGE) {
+                uint64_t end_filter = filter[i++];
+                if (filter_item <= timestamp && timestamp < end_filter) {
+                    match = 1;
+                    break;
+                }
+                
+                continue;
```

Comment:
it would be more logical to make the item-matching queries branch below nested under `else`, instead of `continue` here